.github/workflows/main.yml

```diff
@@ -15,30 +15,41 @@ on:
     branches:
       - '**'
 
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+
 jobs:
   tests:
     name: ${{ matrix.os }} / ${{ matrix.python-version }} ${{ matrix.suffix }}
-    runs-on: ${{ matrix.os }}-latest
-    continue-on-error: ${{ matrix.experimental }}
+    runs-on: ${{ matrix.image }}
     strategy:
       matrix:
-        os: [Ubuntu, MacOS, Windows]
+        os: [Ubuntu, macOS, Windows]
         python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
-        experimental: [false]
         bootstrap-args: [""]
         suffix: [""]
         include:
           - os: Ubuntu
+            image: ubuntu-latest
+          - os: Windows
+            image: windows-2022
+          - os: macOS
+            image: macos-11
+          - os: Ubuntu
+            image: ubuntu-latest
             python-version: "3.10"
-            experimental: true
             bootstrap-args: "--git https://github.com/python-poetry/poetry.git"
             suffix: (HEAD)
           - os: Ubuntu
+            image: ubuntu-latest
             python-version: "3.11-dev"
-            experimental: true
             bootstrap-args: "--git https://github.com/python-poetry/poetry.git"
             suffix: (HEAD)
       fail-fast: false
+    defaults:
```

Comment:
This significantly cleans up our workflows.

Reply:
This adds some toil for us right? Wonder if dependency bot supports bumping these.

Reply:
I think you might have commented too low -- were you talking about the images?

If so, I don't think so. These change rarely, and there is an announcement on the Github Blog.

Reply:
Correct. Ah will check the post. Thanks.